zebrad/tests/acceptance.rs

```diff
@@ -47,7 +47,14 @@ fn generate_no_args() -> Result<()> {
     zebra_test::init();
     let (tempdir, _guard) = tempdir(true)?;
 
-    let child = get_child(&["generate"], &tempdir)?;
+    // Path to generated config file
+    let mut generated_config_path = tempdir.clone();
+    generated_config_path.push("zebrad.toml");
+
+    let child = get_child(
+        &["-c", generated_config_path.to_str().unwrap(), "generate"],
```

Comment:
instead of putting this in the args for every instance of `get_child` lets setup the config args in `get_child` based on the provided `tempdir` and avoid using the `get_child` helper when we need to run commands like `generate` that won't use the standard config path.

```rust
pub fn get_child(args: &[&str], tempdir: &PathBuf) -> Result<TestChild> {
    let mut cmd = test_cmd(env!("CARGO_BIN_EXE_zebrad"), &tempdir)?;

    Ok(cmd
        .args(args)
        .stdout(Stdio::piped())
        .stderr(Stdio::piped())
        .arg("-c")
        .arg(tempdir.join("zebrad.toml"))
        .spawn2()
        .unwrap())
}
```

Reply:
What about adding a `default_config` parameter to `get_child`? If `true` return your code else return current code.

Reply:
I'm not as big of a fan of that approach because then you'll end up with a bunch of `get_child(args, tempdir, true)` calls where the meaning of the boolean is opaque unless you can see the definition and the name of the argument.

If it were up to me though I'd probably have never added `get_child` and I'd instead just use the `std::process::Command` interface directly in the various tests, with the `stdout`/`stderr` lines moved into `test_cmd`. This is why I suggested just making the "default_config: true" case become the body of `get_config` and let the false case be one where you just manaully write it out inline.

If you'd prefer to keep working with the helper I think we should move to a builder style pattern so we can use chained function calls to act like named optional parameters, but this isn't particularly different from the `std::command::Command` interface which is _itself_ a builder for Commands.

Reply:
Thanks for the detailed explication. I applied these at https://github.com/ZcashFoundation/zebra/pull/946/commits/c1a5807a9b7db756a23bf1df2bc0f22f9cea87c2 and https://github.com/ZcashFoundation/zebra/pull/946/commits/ff1c042836a2106b1ccee45f545b9ec02de15322